src/sync-controller.js

```diff
@@ -415,25 +416,57 @@ export default class SyncController extends videojs.EventTarget {
   }
 
   /**
-   * Probe an fmp4 or an mpeg2-ts segment to determine the start of the segment
-   * in it's internal "media time".
+   * Probe an fmp4 segment to determine the start of the segment
+   * in it's internal "composition time", which is equal to the base
+   * media decode time plus the composition time offset value
    *
    * @private
    * @param {SegmentInfo} segmentInfo - The current active request information
-   * @return {object} The start and end time of the current segment in "media time"
+   * @return {object} The start and end time of the current segment in "composition time"
    */
   probeMp4Segment_(segmentInfo) {
-    let segment = segmentInfo.segment;
-    let timescales = mp4probe.timescale(segment.map.bytes);
-    let startTime = mp4probe.startTime(timescales, segmentInfo.bytes);
+    const segment = segmentInfo.segment;
+    // get timescales from init segment
+    const timescales = mp4probe.timescale(segment.map.bytes);
+    // get `traf` boxes from the media segment
+    const trafBoxes = mp4probe.findBox(segmentInfo.bytes, ['moof', 'traf']);
```

Comment:
I feel like this logic would be better suited staying within mux.js and creating a new function like `mp4probe.compositionStartTime`

Reply:
Agreed, done: https://github.com/videojs/mux.js/pull/304